trin-core/src/types/block_body.rs

```diff
@@ -57,6 +47,73 @@ impl Uncles {
     }
 }
 
+impl TryFrom<BlockBodyHelper> for BlockBody {
+    type Error = ssz::DecodeError;
+
+    fn try_from(helper: BlockBodyHelper) -> Result<Self, Self::Error> {
+        let txs: Vec<Transaction> = helper
+            .rlp_txs
+            .iter()
+            .map(|bytes| rlp::decode(bytes).unwrap())
+            .collect();
+        let uncles: Vec<Header> = helper
+            .rlp_uncles
+            .iter()
+            .map(|bytes| rlp::decode(bytes).unwrap())
```

Comment:
We actually don't need to individually encode each uncle header (and probably shouldn't).

Note that the uncles are not included individually, but as a single encoded element (the list is rlp-encoded in aggregate):
https://github.com/ethereum/portal-network-specs/blob/9c7ad893f52e71ecb3e95f81c14710b2253aff51/history-network.md#block-body

I think this is going to give you different SSZ encodings, looking at how it's built below.